test/instrumentation/modules/cassandra-driver/cassandra-driver.test.js

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict';
+
+if (process.env.GITHUB_ACTIONS === 'true' && process.platform === 'win32') {
+  console.log('# SKIP: GH Actions do not support docker services on Windows');
+  process.exit(0);
+}
+
+const test = require('tape');
+const semver = require('semver');
+const { validateSpan } = require('../../../_validate_schema');
+const {
+  runTestFixtures,
+  safeGetPackageVersion,
+  sortApmEvents,
+} = require('../../../_utils');
+const { NODE_VER_RANGE_IITM_GE16 } = require('../../../testconsts');
+
+const CASSANDRA_VERSION = safeGetPackageVersion('cassandra-driver');
+const TEST_KEYSPACE = 'mykeyspace';
+const TEST_TABLE = 'myTable';
+const TEST_DATACENTER = 'datacenter1';
+const TEST_USE_PROMISES = String(semver.satisfies(CASSANDRA_VERSION, '>=3.2'));
+
+const testFixtures = [
+  {
+    name: 'cassandra-driver simple usage for versions <4.7.0',
+    script: 'fixtures/use-cassandra-driver.js',
+    cwd: __dirname,
+    timeout: 20000, // sanity guard on the test hanging
+    maxBuffer: 10 * 1024 * 1024, // This is big, but I don't ever want this to be a failure reason.
+    env: {
+      TEST_KEYSPACE,
+      TEST_TABLE,
+      TEST_DATACENTER,
+      TEST_USE_PROMISES,
+    },
+    versionRanges: {
+      // v4.7.0 is a bad release for node versions <16.9
+      // and we want to test form 14.7 and above
+      node: '<16.9',
+      'cassandra-driver': '>=3.0.0 <4.7.0',
+    },
```

Comment:
This means we aren't testing with recent node and recent cassandra-driver at all, right?

Reply:
The `versionRanges` feature of `runTestFixtures` is perhaps not good enough to support what we want: which is to decide if we want to run or skip based on a *combination* of relevant versions.  

I.e. in general we'd perhaps want to pass in a *function* to return true/false (basically what isCassandraIncompat() is doing).  That makes for a `runTestFixtures` that has options that aren't that nice/elegant to use. Hrm.

Reply:
Well. I did some kind of hack below by cloning the object and overriding the `versionRanges`

This is the snippet placed just before running the tests
```js
// We need to do exactly the same test for `cassandra-driver` v4.7.0 and up
// the only difference is we require NodeJS version to be >16.
// This is necessary because of an issue in the driver that led to
// a change in the node compatibility from >=8 to >=16
//
// The issue: https://datastax-oss.atlassian.net/browse/NODEJS-665
testFixtures.push(
  Object.assign({}, testFixtures[0], {
    name: 'cassandra-driver simple usage for versions >=4.7.0',
    versionRanges: {
      node: '>=16.9',
      'cassandra-driver': '>=4.7.0',
    },
  }),
);
```

Reply:
Ah, I see. Nice. Sounds good to me for now. 

For the future it would perhaps be nice if `runTestFixtures` supported a `versionRanges` (or related name) function that would get the versions and return a boolean for whether to run the tests.

Reply:
That's a good idea :)